src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -85,6 +85,84 @@ public override string ToString()
     }
 }";
 
+        [Fact, WorkItem(32495, "https://github.com/dotnet/roslyn/issues/32495")]
+        public void CheckImplicitObjectInitializerReceiver()
+        {
+            var comp = CreateCompilation(@"
+public class B
+{
+    public B? f2;
+    public B? f3;
+}
+public class C
+{
+    public B? f;
+    static void Main()
+    {
+        new C() { f = { f2 = null, f3 = null }};
+    }
+}", options: WithNonNullTypesTrue());
+
+            comp.VerifyDiagnostics(
+                // (12,25): warning CS8602: Possible dereference of a null reference.
```

Comment:
Perhaps we should consider putting the warning on the `f =` part, and not the `f2` and `f3` parts. I don't think it's immediately obvious why there's an error on `f2` or `f3`.

Reply:
It still won't be obvious, but I adjusted the location a bit ;-)